pkg/auth/handler/webapp/sso_callback.go

```diff
@@ -44,16 +47,25 @@ func (h *SSOCallbackHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	s := session.GetSession(r.Context())
 
 	if state.AccountManagementToken != "" {
+		redirectURL, err := url.Parse("/settings/identity/oauth")
+		if err != nil {
+			panic(err)
+		}
+
 		_, err = h.AccountManagement.FinishAddingIdentityOAuth(s, &accountmanagement.FinishAddingIdentityOAuthInput{
 			Token: state.AccountManagementToken,
 			Query: r.URL.Query().Encode(),
 		})
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusBadRequest)
+			if identity.IsErrDuplicatedIdentity(err) {
+				err = accountmanagement.NewErrAccountManagementDuplicatedIdentity(err)
+			}
```

Comment:
Why do we need to translate error here? Shouldn't this be done by AccountManagementService?